### DISCUSSION TITLE
Playback position in bars/beats

### DISCUSSION THREAD

**Discussion by gloverab** [leolabs/ableton-js#85] [Q&A]
Hey there, and thank you for this incredible ableton implementation.

I feel like I’m missing something, but I haven’t been able to figure out where I can get/observe the Song position in bars/beats. I’m able to get the position in seconds easily, but can’t for the life of me figure out where to access the current position on the grid.

Any tips? Thanks!

**Comment by gloverab** [comment]
Update: I figured it out. `current_song_position` is giving the total number of beats elapsed, not seconds.

Quick formulas I wrote if anyone wants to save themselves the math.

```
let totalBeatsZeroIndex = Math.floor(currentSongTime);
let barZeroIndex = Math.floor(currentSongTime / 4);

let totalBeats = totalBeatsZeroIndex + 1;
let bar = barZeroIndex + 1;
let beat = Math.floor(currentSongTime - 4 * barZeroIndex) + 1;
let division = Math.floor((currentSongTime - totalBeatsZeroIndex) / 0.25) + 1;
```

**Reply by leolabs** [reply]
That's how I did it as well! As long as your song stays in 4/4 time signature, this will work well 👍

**Reply by gloverab** [reply]
Oh yeah, I should have noted for anyone who comes across it - you can use other time signatures by turning those `4`s into variables based off the `signature_numerator` property.

But while I have the man himself here - are there any plans/is it even possible to implement toggling between the browser window and document window? Something like the ability to browse through clips and load them onto tracks/scenes via javascript? I was looking at the ApplicationView class/interfaces and it looks like there aren't any setters in there. Is that a limitation of the ableton control surface api or have you just not dove in on that side yet?

Thanks for everything @leolabs -this is an awesome tool.

**Reply by leolabs** [reply]
Glad you like it! 😊 

You can use the functions available on the `ApplicationView` class to focus different views. For example, you could focus the browser like this:

```ts
await ableton.application.view.focusView("Browser")
```

There's also `toggleBrowse` which may or may not be better for what you're trying to do. Let me know if this helps!
